server/claim/claim.controller.ts

```diff
@@ -248,26 +248,31 @@ export class ClaimController {
     public async personalityClaimPage(@Req() req: Request, @Res() res: Response) {
         const parsedUrl = parse(req.url, true);
         // @ts-ignore
+        try{
+            const personality = await this.personalityService.getBySlug(
+                req.params.personalitySlug,
+                // @ts-ignore
+                req.language
+            );
 
-        const personality = await this.personalityService.getBySlug(
-            req.params.personalitySlug,
-            // @ts-ignore
-            req.language
-        );
-
-        const claim = await this.claimService.getByPersonalityIdAndClaimSlug(
-            personality._id,
-            req.params.claimSlug
-        );
-
-        await this.viewService
-            .getNextServer()
-            .render(
-                req,
-                res,
-                "/claim-page",
-                Object.assign(parsedUrl.query, { personality, claim })
+            const claim = await this.claimService.getByPersonalityIdAndClaimSlug(
+                personality._id,
+                req.params.claimSlug
             );
+
+            await this.viewService
+                .getNextServer()
+                .render(
+                    req,
+                    res,
+                    "/claim-page",
+                    Object.assign(parsedUrl.query, { personality, claim })
+                );
+        } catch {
```

Comment:
Nice, you find a way of solving it!

I would suggest that, instead of executing next server to render the 404 page, you should raise a NotFoundException and let the [exception filter execute a redirect to 404.](https://github.com/AletheiaFact/aletheia/blob/develop/server/filters/not-found.filter.ts#L7)

Also, you should raise the exception in the service, not in the controller.